src/Parser.js

```diff
@@ -153,7 +153,7 @@ export default class Parser {
         return parse;
     }
 
-    static endOfExpression = ["}", "\\end", "\\right", "&"];
+    static endOfExpression = ["}", "\\endgroup", "\\end", "\\right", "&"];
```

Comment:
Why isn't `]` in this list?  Are the contents of an optional group an expression?

Reply:
It sounds like after we see a `[` then we treat `]` specially.